src/app/OperationalDeviceProxyPool.h

```diff
@@ -61,7 +61,11 @@ class OperationalDeviceProxyPool : public OperationalDeviceProxyPoolDelegate
         return mDevicePool.CreateObject(params, peerId, nodeResolutionData);
     }
 
-    void Release(OperationalDeviceProxy * device) override { mDevicePool.ReleaseObject(device); }
+    void Release(OperationalDeviceProxy * device) override
+    {
+        device->Disconnect();
```

Comment:
So one issue with doing this is that if a binding is removed we will destroy the CASE session even if it's also being used by a subscription or something like that....

(That said, I am not sure those would ever end up reusing the same CASE session given the way things are set up right now...)

We really need to get our lifetime stories for this stuff in order.  :(

This is _probably_ OK, but I'd give it like a 40% chance that it will break something.

Reply:
For the time being, not explicitly calling this Disconnect since it seems `ExpireAllPairings` is called upon every new CASE session establishment. See https://github.com/project-chip/connectedhomeip/issues/15986.